ts/features/pn/analytics/index.ts

```diff
@@ -0,0 +1,110 @@
+import * as O from "fp-ts/lib/Option";
+import * as pot from "@pagopa/ts-commons/lib/pot";
+import { mixpanelTrack } from "../../../mixpanel";
+import { TransactionSummaryErrorContent } from "../../../screens/wallet/payment/NewTransactionSummaryScreen";
+import { PnActivationState } from "../../pn/store/reducers/activation";
+import { PNMessage } from "../../pn/store/types/types";
+
+export function trackPNAttachmentDownloadFailure() {
+  void mixpanelTrack("PN_ATTACHMENT_DOWNLOADFAILURE");
+}
+
+export function trackPNAttachmentOpen() {
+  void mixpanelTrack("PN_ATTACHMENT_OPEN");
+}
+
+export function trackPNAttachmentPreviewStatus(
+  previewStatus: "displayed" | "error"
+) {
+  void mixpanelTrack("PN_ATTACHMENT_PREVIEW_STATUS", {
+    previewStatus
+  });
+}
+
+export function trackPNAttachmentSave() {
+  void mixpanelTrack("PN_ATTACHMENT_SAVE");
+}
+
+export function trackPNAttachmentShare() {
+  void mixpanelTrack("PN_ATTACHMENT_SHARE");
+}
+
+export function trackPNDisclaimerAccepted(
+  messageCreatedAt: Date,
+  messageReceiptDate: Date
+) {
+  void mixpanelTrack("PN_DISCLAIMER_ACCEPTED", {
+    eventTimestamp: new Date().toISOString(),
+    messageTimestamp: messageCreatedAt.toISOString(),
+    notificationTimestamp: messageReceiptDate.toISOString()
+  });
+}
+
+export function trackPNDisclaimerRejected() {
+  void mixpanelTrack("PN_DISCLAIMER_REJECTED");
+}
+
+export function trackPNDisclaimerShowSuccess() {
+  void mixpanelTrack("PN_DISCLAIMER_SHOW_SUCCESS");
+}
+
+export function trackPNNotificationLoadError(errorCode?: string) {
+  const properties = errorCode ? { errorCode } : { jsonDecodeFailed: true };
+  void mixpanelTrack("PN_NOTIFICATION_LOAD_ERROR", properties);
+}
+
+export function trackPNNotificationLoadSuccess(pnMessage: PNMessage) {
+  const notificationStatusHistory = pnMessage.notificationStatusHistory;
+  const lastNotificationIndex = notificationStatusHistory.length - 1;
+  const lastNotification = notificationStatusHistory[lastNotificationIndex];
+  const lastNotificationStatus = lastNotification.status;
+  const messageAttachments = pnMessage.attachments;
+  const messageAttachmentLength = messageAttachments?.length ?? 0;
+  const messageHasAttachments = messageAttachmentLength > 0;
```

Comment:
What do you think about
```
pipe(
  pnMessage.notificationStatusHistory as Array<NotificationStatusHistoryElement>,
  A.last,
  O.map(lastNotification => lastNotification.status),
  O.fold(
    () => undefined,
    (status: string) => void mixpanelTrack("PN_NOTIFICATION_LOAD_SUCCESS", {
      notificationLastStatus: status,
      hasAttachments: pipe(
        pnMessage.attachments,
        O.fromNullable,
        O.map(messageAttachments => messageAttachments.length > 0),
        O.fold(
          () => false,
          identity
        )
      )
    })
  )
);
```

it should also cover the previously-uncovered case where the input is empty (by reporting nothing - instead of throwing an error)

Reply:
That's fine by me anyway we could modify `hasAttachament` like this:
```Typescript
pipe(
  pnMessage.attachments,
  O.fromNullable,
  O.map(A.isNonEmpty),
  O.getOrElse(() => false)
)
```

Reply:
Applied